jobbergate-api/jobbergate_api/apps/job_submissions/routers.py

```diff
@@ -300,6 +300,8 @@ async def job_submissions_agent_pending(
                 job_submissions_table.c.id,
                 job_submissions_table.c.job_submission_name,
                 job_submissions_table.c.job_submission_owner_email,
+                job_submissions_table.c.execution_directory,
+                job_submissions_table.c.execution_parameters,
```

Comment:
something I noticed from my notes is the fact that the execution directory is a execution parameter from slurmrestd. For example, here it follows a slurmrestd payload that works:

```json
{
    "script": "#!/bin/bash\nsrun hostname",
    "job": {
        "name": "test-6",
        "environment": {
            "PATH": "/bin:/usr/bin/:/usr/local/bin/",
            "LD_LIBRARY_PATH": "/lib/:/lib64/:/usr/local/lib"
        },
        "partition": "osd-slurmd",
        "nodes": [
            47,
            48
        ],
        "current_working_directory": "/tmp/"
    }
}
```

In this case, why do not make the execution directory just a mandatory field from the execution params instead of making it to have a specific column?

Reply:
This logic is done on cluster-agent, notice some job parameters are set by default and can be overwritten by user-supplied ones:
https://github.com/omnivector-solutions/cluster-agent/blob/30a0399f8ddbb6d92f36c28f6e0ff37ea96fa8a4/cluster_agent/jobbergate/submit.py#L109-L115 

I agree with our suggestion, but we need to investigate the impact on the users' workflow and make sure we have backward compatibility. That could be covered by a follow-up task.

Reply:
yes, agreed